src/main/scala/temple/generate/server/CreatedByAttribute.scala

```diff
@@ -2,8 +2,38 @@ package temple.generate.server
 
 import temple.ast.AttributeType
 
-case class CreatedByAttribute(
-  inputName: String,
-  name: String,
-  attributeType: AttributeType,
-)
+sealed trait CreatedByAttribute
```

Comment:
I know this has taken quite a while, but it feels like `Option[CreatedByAttribute]` (where `case class CreatedByAttribute(inputName: String, name: String, attributeType: AttributeType, filterEnumeration: Boolean)`, would be better, given that

1) There’s no difference between the arguments of `EnumerateByCreator` and `EnumerateByAll`, either in storage or semantics.
2) The switching between them is only ever done in a boolean manner.
3) CreatedByAttribute.None is serving the exact purpose of Option.None.

Would be interested to hear @jaylees14’s thoughts too

Reply:
Yeah I'm okay with that 👍

Reply:
The reason we went with this is because we could feasibly add different enumerations in the future

Reply:
Discussed offline, it wasn't worth the time to switch